ome/ome-model
# xsd-fu: Make conversion checks work for C++

- Correct the Java checks
- Make the checks work for C++
- Make errors fatal so we catch them (the C++ issues weren't noticed until I started working on the model a few days back)

This adds additional regexes to pick the names from the C++ sources.  It also corrects the logic for determining the conversion filename since this isn't generated, and for C++ we need to distinguish between source and header generation (we run the check only when generating headers since the checks are against the generated headers, plus the static conversion file).  This wasn't an issue for Java.

We also amend the checks to look at the differences between these sets:

```
    (convert) - (store) | (retrieve)
    (store) | (retrieve) - (convert)
```

The existing logic was comparing store and retrieve sets, but we really want to pick up on methods which are in store *or* retrieve, but which are missing in convert.

This change means we have full checking for both the Java and C++ builds at all times.  Anyone changing any of these files will be required to update the interfaces and conversion logic in lockstep for the build to remain green, so this will prevent any inconsistency creeping in.

Testing:

- check jobs remain green
- remove a conversion block from `ome-xml/src/main/java/ome/xml/meta/MetadataConverter.java` and run the maven build (or `mvn generate-sources`).  It should fail with a diagnostic indicating the missing methods during the generation step.
- remove a conversion block from `ome-xml/src/main/cpp/ome/xml/meta/Convert.cpp` and run the C++ build (or just the `gensrc` target).  It should fail with a diagnostic indicating the missing methods during the source generation step.  Basically, run the superbuild with this branch checked out, then run the `ome-model` target.
